src/modules/cpu/kernel/noise_gaussian.hpp

```diff
@@ -1392,3 +1392,499 @@ RppStatus gaussian_noise_i8_i8_host_tensor(Rpp8s *srcPtr,
+{
+    RpptROI3D roiDefault;
+    if(srcGenericDescPtr->layout==RpptLayout::NCDHW)
+        roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[4], (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[2]};
+    else if(srcGenericDescPtr->layout==RpptLayout::NDHWC)
+        roiDefault = {0, 0, 0, (Rpp32s)srcGenericDescPtr->dims[3], (Rpp32s)srcGenericDescPtr->dims[2], (Rpp32s)srcGenericDescPtr->dims[1]};
+    Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < batchSize; batchCount++)
+    {
+        RpptROI3D roi;
+        RpptROI3DPtr roiPtrInput = &roiGenericPtrSrc[batchCount];
+        compute_roi3D_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32f *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrImage = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32f mean = meanTensor[batchCount];
+        Rpp32f stdDev = stdDevTensor[batchCount];
+        Rpp32u offset = batchCount * srcGenericDescPtr->strides[0];
+        Rpp32u bufferLength = roi.xyzwhdROI.roiWidth * layoutParams.bufferMultiplier;
+        bool copyInput = (!mean) && (!stdDev);
+        if (copyInput)
+        {
+            copy_3d_host_tensor(srcPtrImage, srcGenericDescPtr, dstPtrImage, dstGenericDescPtr, &roi, layoutParams);
+        }
+        else
+        {
+            Rpp32f *srcPtrChannel, *dstPtrChannel;
+            dstPtrChannel = dstPtrImage;
+            RpptXorwowStateBoxMuller xorwowState;
+#if __AVX2__
+            Rpp32u alignedLength = (bufferLength / 24) * 24;
+            Rpp32u vectorIncrement = 24;
+            Rpp32u vectorIncrementPerChannel = 8;
+            __m256i pxXorwowStateX[5], pxXorwowStateCounter;
+            rpp_host_rng_xorwow_state_offsetted_avx(xorwowInitialStatePtr, xorwowState, offset, pxXorwowStateX, &pxXorwowStateCounter);
+
+            __m256 pGaussianNoiseParams[2];
+            pGaussianNoiseParams[0] = _mm256_set1_ps(mean);
+            pGaussianNoiseParams[1] = _mm256_set1_ps(stdDev);
+#else
+            Rpp32u alignedLength = (bufferLength / 12) * 12;
+            Rpp32u vectorIncrement = 12;
+            Rpp32u vectorIncrementPerChannel = 4;
+            __m128i pxXorwowStateX[5], pxXorwowStateCounter;
+            rpp_host_rng_xorwow_state_offsetted_sse(xorwowInitialStatePtr, xorwowState, offset, pxXorwowStateX, &pxXorwowStateCounter);
+
+            __m128 pGaussianNoiseParams[2];
+            pGaussianNoiseParams[0] = _mm_set1_ps(mean);
+            pGaussianNoiseParams[1] = _mm_set1_ps(stdDev);
+#endif
+            // Gaussian Noise without fused output-layout toggle (NDHWC -> NDHWC)
+            if((srcGenericDescPtr->dims[4] == 3) && (srcGenericDescPtr->layout == RpptLayout::NDHWC) && (dstGenericDescPtr->layout == RpptLayout::NDHWC))
```

Comment:
do we have test to bit compare the AVX output with c reference output. This is hard to review for correctness